e2e/volumegroupsnapshot.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2024 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	groupsnapapi "github.com/kubernetes-csi/external-snapshotter/client/v8/apis/volumegroupsnapshot/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+type cephFSVolumeGroupSnapshot struct {
+	*volumeGroupSnapshotterBase
+}
+
+var _ VolumeGroupSnapshotter = &cephFSVolumeGroupSnapshot{}
+
+func newCephFSVolumeGroupSnapshot(f *framework.Framework, namespace,
+	storageClass string,
+	blockPVC bool,
+	timeout, totalPVCCount int,
+) (VolumeGroupSnapshotter, error) {
+	base, err := newVolumeGroupSnapshotBase(f, namespace, storageClass, blockPVC, timeout, totalPVCCount)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create volumeGroupSnapshotterBase: %w", err)
+	}
+
+	return &cephFSVolumeGroupSnapshot{
+		volumeGroupSnapshotterBase: base,
+	}, nil
+}
+
+func (c *cephFSVolumeGroupSnapshot) TestVolumeGroupSnapshot() error {
+	return c.volumeGroupSnapshotterBase.testVolumeGroupSnapshot(c)
```

Comment:
Why not just export `volumeGroupSnapshotterBase.TestVolumeGroupSnapshot()` ? You do not need to have this function at all then.

Reply:
thats not possible as `testVolumeGroupSnapshot` internal method expects an input of struct to run different validator to run , that why this is a small wraper to run Test from higher functions by not exposing the details of the test.

Reply:
ok, it might be able to restructure things a little more and have it work, but that isn't required as this works well too.